File: src/Lesson5/java/Ex8.java
package Lesson5.java;
import java.util.Arrays;

public class Ex8 {

    public static void reverseArray(int[] arr) {

        int left = 0;
        int right = arr.length - 1;

        while (left < right) {

            int temp = arr[left];
            arr[left] = arr[right];
            arr[right] = temp;

            left++;
            right--;
        }
    }

    public static void main(String[] args) {

        int[] arr = {1, 2, 3, 4, 5};

        System.out.println("Original array: " + Arrays.toString(arr));

        reverseArray(arr);

        System.out.println("Reversed array: " + Arrays.toString(arr));
    }
}

File: src/Lesson5/forkotlin/Ex8.kt
package Lesson5.forkotlin

fun reverseArray(arr: IntArray) {

    var left = 0
    var right = arr.size - 1

    while (left < right) {

        val temp = arr[left]
        arr[left] = arr[right]
        arr[right] = temp

        left++
        right--
    }
}

fun main() {

    val arr = intArrayOf(1, 2, 3, 4, 5)

    println("Original array: ${arr.joinToString()}")

    reverseArray(arr)

    println("Reversed array: ${arr.joinToString()}")
}